docs/sources/developers/plugins/migration-guide.md

```diff
@@ -4,77 +4,451 @@ title = "Plugin migration guide"
+## Introduction
+
+This guide helps you identify the steps you need to take based on the Grafana version your plugin supports and explains how to migrate the plugin to the 8.2.x or a later version.
+
+> **Note:** If you've successfully migrated your plugin using this guide, then share your experiences with us! If you find missing information, then we encourage you to [submit an issue on GitHub](https://github.com/grafana/grafana/issues/new?title=Docs%20feedback:%20/developers/plugins/migration-guide.md) so that we can improve this guide!
+
+## Table of contents
+
+- [From version 7.x.x to 8.0.0](#from-version-7xx-to-800)
+  - [Backend plugin v1 support has been dropped](#backend-plugin-v1-support-has-been-dropped)
+    - [1. Add dependency on grafana-plugin-sdk-go](#1-add-dependency-on-grafana-plugin-sdk-go)
+    - [2. Update the way you bootstrap your plugin](#2-update-the-way-you-bootstrap-your-plugin)
+    - [3. Update the plugin package](#3-update-the-plugin-package)
+  - [Unsigned backend plugins will not be loaded](#unsigned-backend-plugins-will-not-be-loaded)
+  - [Time series data can now be in wide or many format](#time-series-data-can-now-be-in-wide-or-many-format)
+  - [Update react-hook-form from v6 to v7](#update-react-hook-form-from-v6-to-v7)
+  - [Update the plugin.json](#update-the-pluginjson)
+  - [Update imports to match emotion 11](#update-imports-to-match-emotion-11)
+  - [8.0 Deprecations](#80-deprecations)
+    - [Grafana theme v1](#grafana-theme-v1)
+- [From version 6.2.x to 7.4.0](#from-version-62x-to-740)
+  - [Legend components](#legend-components)
+- [From version 6.x.x to 7.0.0](#from-version-6xx-to-700)
+  - [What's new in Grafana 7.0?](#whats-new-in-grafana-70)
+  - [Migrate a plugin from Angular to React](#migrate-a-plugin-from-angular-to-react)
+    - [Migrate a panel plugin](#migrate-a-panel-plugin)
+    - [Migrate a data source plugin](#migrate-a-data-source-plugin)
+    - [Migrate to data frames](#migrate-to-data-frames)
+  - [Troubleshoot plugin migration](#troubleshoot-plugin-migration)
+
+## From version 7.x.x to 8.x.x
+
+This section explains how to migrate Grafana v7.x.x plugins to the updated plugin system available in Grafana v8.x.x. Depending on your plugin, you need to perform one or more of the following steps. We have documented the breaking changes in Grafana v8.x.x and the steps you need to take to upgrade your plugin.
+
+### Backend plugin v1 support has been dropped
+
+Use the new [plugin sdk](https://github.com/grafana/grafana-plugin-sdk-go) to run your backend plugin running in Grafana 8.
+
+#### 1. Add dependency on grafana-plugin-sdk-go
+
+Add a dependency on the `https://github.com/grafana/grafana-plugin-sdk-go`. We recommend using [go modules](https://go.dev/blog/using-go-modules) to manage your dependencies.
+
+#### 2. Update the way you bootstrap your plugin
```

Comment:
I love this in-depth explainer, with code samples (!!!), of how to migrate. Perhaps it could be better suited to a subpage (`/migration-guide/updating-to-plugin-sdk` or something), with a brief summary of the change, and why.

Reply:
Thanks @joshhunt!

Regarding the why - I think the main reason behind the plugin SDK was to reduce the boilerplate and make it easier to just spin up a backend plugin. Will check a bit more though on what our direction is with the SDK.

The subpage is not a bad idea, we are also thinking on how we could better organise our migration guides (and also the other plugin related docs). We probably won't do it in this PR, but will let you and the other frontenders as soon as we have a design doc.